lib/src/arithmetic/floating_point/floating_point_converter.dart

```diff
@@ -0,0 +1,420 @@
+  ///
+  /// 1. The exponent is adjusted according to the given [destExponentWidth].
+  /// 2. The mantissa is adjusted according to the given [destMantissaWidth] and
+  ///    the rounding mode [FloatingPointRoundingMode.roundNearestEven].
+  /// 3. If the exponent is all ones, the output [FloatingPoint] is an infinity
+  ///    with the same sign as the [source].
+  /// 4. If the exponent is all zeros, the output [FloatingPoint] is a zero with
+  ///    the same sign as the [source].
+  /// 5. Otherwise, the output [FloatingPoint] is the result of the adjusted
+  ///    exponent and the adjusted mantissa.
+  static FloatingPoint convertNormalNumber(
+      {required FloatingPoint source,
+      required int destExponentWidth,
+      required int destMantissaWidth}) {
+    final adjustedExponent = adjustExponent(source.exponent, destExponentWidth);
+
+    final adjustedMantissa =
+        Logic(name: 'adjustedMantissa', width: destMantissaWidth);
+
+    adjustedMantissa <=
+        adjustMantissaPrecision(source.mantissa, destMantissaWidth,
+            Const(FloatingPointRoundingMode.roundNearestEven.index));
+
+    final isOverflow = adjustedExponent
+        .gt(FloatingPointValue.computeMaxExponent(destExponentWidth));
+    final isUnderflow = adjustedExponent.lt(0);
+
+    final packNormal = FloatingPoint(
+        exponentWidth: destExponentWidth, mantissaWidth: destMantissaWidth);
+
+    print("isOverflow:");
+    print(isOverflow.value.toString());
+
+    Combinational([
+      If.block([
+        Iff(isOverflow, [
+          packNormal <
+              handleOverflow(
+                  source: source,
+                  destExponentWidth: destExponentWidth,
+                  destMantissaWidth: destMantissaWidth),
+        ]),
+        ElseIf(isUnderflow, [
+          packNormal <
+              handleUnderflow(
+                  source: source,
+                  destExponentWidth: destExponentWidth,
+                  destMantissaWidth: destMantissaWidth),
+        ]),
+        Else([
+          packNormal.sign < source.sign,
+          packNormal.exponent < adjustedExponent,
+          packNormal.mantissa < adjustedMantissa
+        ])
+      ]),
+    ]);
+
+    print(packNormal.value.toString());
+    return packNormal;
+  }
+
+  Logic _normalizeSubnormalExponent() =>
+      Const(0, width: destExponentWidth, fill: true);
+  Logic _normalizeSubnormalMantissa() =>
+      Const(0, width: destMantissaWidth, fill: true);
+
+  /// Handle the case where the [source] is too large to be represented by
+  /// a [FloatingPoint] with the given [destExponentWidth] and
+  /// [destMantissaWidth].
+  ///
+  /// The resulting [FloatingPoint] is a non-NaN infinity with the same sign as
+  /// the [source].
+  ///
+  /// [destExponentWidth] and [destMantissaWidth] are the widths of the exponent
+  /// and mantissa of the destination [FloatingPoint] respectively.
+  static FloatingPoint handleOverflow(
```

Comment:
you may want to tag these helper functions with `@visibleForTesting` unless you think they are useful outside of this converter?